lib/msf/core/payload_set.rb

```diff
@@ -427,6 +458,14 @@ def build_payload(*modules)
     return klass
   end
 
+  def build_adapted_name(aname, pname)
+    aparts = aname.split('/')
+    pparts = pname.split('/')
+    pparts.shift if aparts.last.start_with?(pparts.first)
```

Comment:
This removes the redundant `python` part of the adapted payload name, while enabling us in the future to have new subtypes like `python_technique1` / `python_technique2` etc.